pkg/rewards/7_goldActiveODRewards.go

```diff
@@ -0,0 +1,154 @@
+    FROM od_rewards_with_block_info
+        WHERE end_timestamp >= TIMESTAMP '{{.rewardsStart}}'
+		and start_timestamp <= TIMESTAMP '{{.cutoffDate}}'
+		and block_time <= TIMESTAMP '{{.cutoffDate}}' -- Always ensure we're not using future data. Should never happen since we're never backfilling, but here for safety and consistency.
+),
+-- Cut each reward's start and end windows to handle the global range
+active_rewards_updated_end_timestamps as (
+ SELECT
+	 avs,
+	 operator,
+	 /**
+	  * Cut the start and end windows to handle
+	  * A. Retroactive rewards that came recently whose start date is less than start_timestamp
+	  * B. Don't make any rewards past end_timestamp for this run
+	  */
+	 start_timestamp as reward_start_exclusive,
+	 LEAST(global_end_inclusive, end_timestamp) as reward_end_inclusive,
+	 tokens_per_day,
+	 token,
+	 multiplier,
+	 strategy,
+	 reward_hash,
+	 global_end_inclusive,
+	 block_date as reward_submission_date
+ FROM active_rewards_modified
+),
+-- For each reward hash, find the latest snapshot
+active_rewards_updated_start_timestamps as (
+	SELECT
+		ap.avs,
+		ap.operator,
+		COALESCE(MAX(g.snapshot), ap.reward_start_exclusive) as reward_start_exclusive,
+		-- ap.reward_start_exclusive,
+		ap.reward_end_inclusive,
+		ap.token,
+		-- We use floor to ensure we are always underesimating total tokens per day
+		floor(ap.tokens_per_day) as tokens_per_day_decimal,
+		-- Round down to 15 sigfigs for double precision, ensuring know errouneous round up or down
+		ap.tokens_per_day * ((POW(10, 15) - 1)/(POW(10, 15))) as tokens_per_day,
+		ap.multiplier,
+		ap.strategy,
+		ap.reward_hash,
+		ap.global_end_inclusive,
+	 	ap.reward_submission_date
+	FROM active_rewards_updated_end_timestamps ap
+	LEFT JOIN gold_table g ON g.reward_hash = ap.reward_hash
+ GROUP BY ap.avs, ap.operator, ap.reward_end_inclusive, ap.token, ap.tokens_per_day, ap.multiplier, ap.strategy, ap.reward_hash, ap.global_end_inclusive, ap.reward_start_exclusive, ap.reward_submission_date
+),
+-- Parse out invalid ranges
+active_reward_ranges AS (
+	SELECT * from active_rewards_updated_start_timestamps
+	/** Take out (reward_start_exclusive, reward_end_inclusive) windows where
+	* 1. reward_start_exclusive >= reward_end_inclusive: The reward period is done or we will handle on a subsequent run
+	*/
+	WHERE reward_start_exclusive < reward_end_inclusive
+),
+-- Explode out the ranges for a day per inclusive date
+exploded_active_range_rewards AS (
+	SELECT
+	    *
+	FROM active_reward_ranges
+	CROSS JOIN generate_series(DATE(reward_start_exclusive), DATE(reward_end_inclusive), INTERVAL '1' DAY) AS day
+),
+active_rewards_final AS (
+	SELECT
+		avs,
+		operator,
+		cast(day as DATE) as snapshot,
+		token,
+		tokens_per_day,
+		tokens_per_day_decimal,
+		multiplier,
+		strategy,
+		reward_hash,
+		reward_submission_date
+	FROM exploded_active_range_rewards
+	-- Remove snapshots on the start day
```

Comment:
I actually don't think we should combine it. We should keep this separate. Because the amounts mean different things in this reward type. In the original rewards, it means the amount for the entire reward submission, but here the it's an operator amount.